### PR TITLE
Add SO-16.

### PR DESCRIPTION
**Pull request by jmgao**
Datasheet: https://www.nxp.com/docs/en/package-information/SOT109-1.pdf
Screenshot:
so-16

### CODE REVIEW

**Review by chschlue — CHANGES_REQUESTED** [review]
See line comments.
Also, please add 1.75mm `overall_height`

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/so.yaml [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
-    
+
+SO-16_3.9x9.9mm_P1.27mm:
+  size_source: 'https://www.nxp.com/docs/en/package-information/SOT109-1.pdf'
+  body_size_x:
+    minimum: 3.8
+    maximum: 4.0
+  body_size_y:
+    minimum: 9.8
+    maximum: 10.0
```
[the remark this answers is not in the document]
Comment (jmgao) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/so.yaml]: Done.

File: scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/so.yaml [hunk 2]
```diff
@@ -1,7 +1,28 @@
 FileHeader:
   library_Suffix: 'SO'
   device_type: 'SO'
-    
+
+SO-16_3.9x9.9mm_P1.27mm:
+  size_source: 'https://www.nxp.com/docs/en/package-information/SOT109-1.pdf'
+  body_size_x:
+    minimum: 3.8
+    maximum: 4.0
```
[the remark this answers is not in the document]
Comment (jmgao) [reply scripts/Packages/Package_Gullwing__QFP_SOIC_SO/size_definitions/so.yaml]: Done.

**Review by chschlue — APPROVED** [review]
Thanks

**Pull request merged**